core/src/main/java/org/everit/json/schema/CombinedSchema.java

```diff
@@ -191,10 +194,22 @@ public ValidationCriterion getCriterion() {
         return criterion;
     }
 
+    /**
+     * Returns the subschemas in the order they were added.
+     * @return the subschemas in insertion order
+     */
     public Collection<Schema> getSubschemas() {
         return subschemas;
     }
 
+    /*
+     * Internal method that returns the subschemas in the order they should be visited
+     * by the ValidatingVisitor and the equals and hashCode methods.
+     */
+    protected Collection<Schema> visitSubschemas() {
```

Comment:
nit: I think package-private visibility would be better here, so that potential subclasses outside the package don't see it. It would also mean, we wouldn't need to maintain backward compatibility in the future - just in case we change our mind.

Also some little more descriptive would be welcome. How about `subschemasWithCombinedFirst()` ?

Reply:
Thanks, incorporated your suggestions.